common/src/main/java/com/wynntils/features/MountHorseHotkeyFeature.java

```diff
@@ -0,0 +1,186 @@
+        if (McUtils.player().getVehicle() != null) {
+            postHorseErrorMessage(MountHorseStatus.ALREADY_RIDING);
+            return;
+        }
+
+        Entity horse = searchForHorseNearby();
+        if (horse == null) { // Horse has not spawned, we should do that
+            int horseInventorySlot = findHorseInInventory();
+            if (horseInventorySlot > 8 || horseInventorySlot == -1) {
+                postHorseErrorMessage(MountHorseStatus.NO_HORSE);
+                return;
+            }
+            trySummonAndMountHorse(horseInventorySlot, summonAttempts);
+        } else { // Horse already exists, mount it
+            mountHorse(horse);
+        }
+    });
+
+    /** Horse should be nearby when this is called */
+    private static void mountHorse(Entity horse) {
+        Player player = McUtils.player();
+
+        // swap to soul points to avoid right click problems
+        int prevItem = player.getInventory().selected;
+        player.getInventory().selected = 8;
+        McUtils.player()
+                .connection
+                .send(ServerboundInteractPacket.createInteractionPacket(horse, false, InteractionHand.MAIN_HAND));
+        player.getInventory().selected = prevItem;
+    }
+
+    private static void trySummonAndMountHorse(int horseInventorySlot, int attempts) {
+        if (attempts <= 0) {
+            postHorseErrorMessage(MountHorseStatus.NO_HORSE);
+            return;
+        }
+
+        Player player = McUtils.player();
+        if (!alreadySetPrevItem) {
+            prevItem = player.getInventory().selected;
+            alreadySetPrevItem = true;
+        }
+
+        new Delay(
+                () -> {
+                    player.getInventory().selected = horseInventorySlot;
+                    McUtils.player().connection.send(new ServerboundUseItemPacket(InteractionHand.MAIN_HAND));
+
+                    Entity horse = searchForHorseNearby();
+                    if (isPlayersHorse(horse)) { // Horse successfully summoned
+                        player.getInventory().selected = prevItem;
+                        alreadySetPrevItem = false;
+                        new Delay(() -> mountHorse(horse), 1);
+                        return;
+                    }
+                    trySummonAndMountHorse(horseInventorySlot, attempts - 1);
+                },
+                summonDelayTicks);
+    }
+
+    private static boolean isPlayersHorse(Entity horse) {
+        if (!(horse instanceof AbstractHorse)) return false;
+
+        String playerName = McUtils.player().getName().getString();
+        String defaultName = "§f" + playerName + "§7" + "'s horse";
+        String customNameSuffix = "§7" + " [" + playerName + "]";
+
+        Component horseName = horse.getCustomName();
+        if (horseName == null) return false;
+
+        return defaultName.equals(horseName.getString())
+                || horseName.getString().endsWith(customNameSuffix);
+    }
```

Comment:
this technically should be abstracted away into another folder, but this shouldn't matter for now